test/blackbox-tests/test-cases/macro-expand-error.t/dune

```diff
@@ -1 +0,0 @@
-(copy_files %{read:x}/*)
```

Comment:
I'm wondering if this limitation should only be lifted for >= 3.0

Reply:
It should. I have a reminder about this: https://github.com/ocaml/dune/issues/4335

Reply:
I ended up reverting this change after using a different approach in `expander.ml`. It would still possible (and easy) to re-allow `%{read:...}` in this position, but it makes more sense to do it in a separate PR.